lib/ansible/galaxy/api.py

```diff
@@ -696,17 +697,26 @@ def wait_import_task(self, task_id, timeout=0):
         start = time.time()
         wait = 2
 
+        token_expired = False
         while timeout == 0 or (time.time() - start) < timeout:
             try:
                 data = self._call_galaxy(full_url, method='GET', auth_required=True,
                                          error_context_msg='Error when getting import task results at %s' % full_url)
             except GalaxyError as e:
-                if e.http_code != 404:
+                if e.http_code == 401 and self.token and isinstance(self.token, KeycloakToken):
+                    if token_expired:
```

Comment:
Is this a check just for the second failed iteration in a row? Maybe have a counter to represent this?

Reply:
I could make it a counter, but unless we attempt refreshing the token more than once in a row, I think the boolean reads a little better.

Reply:
Seems like with https://github.com/ansible/ansible/pull/79145/files/0c2d16d6eaa97b5e1dc95cbddb199acac5aa0d74#r1013488315 refreshing would be redundant.

Reply:
Yep, just haven't removed this yet.